### ISSUE TITLE
[commonjs] Undeclared value gets exported

### ISSUE THREAD

**Issue by bpierre** [rollup/plugins#532]
- Rollup Plugin Name: `commonjs`
- Rollup Plugin Version: `14.0.0`
- Rollup Version: `2.23.0`
- Operating System (or Browser): Linux
- Node Version: 13.7.0
- Link to reproduction: https://gist.github.com/bpierre/981a9c9d79adc32f06c4bcf9fef3962c

### How to reproduce:

```console
git clone git@gist.github.com:981a9c9d79adc32f06c4bcf9fef3962c.git
cd 981a9c9d79adc32f06c4bcf9fef3962c
yarn && yarn rollup -c
cat dist/981a9c9d79adc32f06c4bcf9fef3962c/node_modules/@walletconnect/web3-provider/dist/cjs/index.js
```

### Expected Behavior

The last line of the output shouldn’t contain `index as i`:

```js
// …
export { cjs as __moduleExports };
```

### Actual Behavior

An non-declared value, `index`, gets exported:

```js
// …
export { cjs as __moduleExports, index as i };
```

**Comment by shellscape** [comment]
Thanks for opening an issue. Citing the issue template:

>   🚨 Issues WITHOUT a valid reproduction WILL BE CLOSED!
> 
>   Please provide one by:
>   1. Using the REPL.it plugin reproduction template at https://repl.it/@rollup/rollup-plugin-repro
>   2. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
>      Please use NPM for installing dependencies!
>      These may take more time to triage than the other options.
>   3. Using the Rollup REPL at https://rollupjs.org/repl/
> 
>   ⚠️ ZIP Files are unsafe and maintainers will NOT download them.

We cannot make this any clearer. Please add a valid reproduction from the list above and we'll be happy to triage further.

**Comment by shellscape** [comment]
Hey sorry, the script got ya. It doesn't recognize a gist as a repo. It's a little unorthodox, but no worries.

**Comment by lukastaegert** [comment]
It appears that this is a bug in Rollup core that is triggered by dynamically importing a file with synthetic named exports: https://repl.it/@lukastaegert/rollup-plugin-repro-1#dist/lib.js

**Comment by lukastaegert** [comment]
Fix at rollup/rollup#3738
